doc/deprecated.rst

```diff
@@ -296,3 +296,23 @@ Node
 * Instantiating ``Twig\Node\Node`` directly is deprecated as of Twig 3.15. Use
   ``EmptyNode`` or ``Nodes`` instead depending on the use case. The
   ``Twig\Node\Node`` class will be abstract in Twig 4.0.
+
+Operators
+---------
+
+* Using ``~`` with ``+`` or ``-`` in an expression without using parentheses to
+  clarify precedence is deprecated as of Twig 3.15 (in Twig 4.0, parentheses
+  won't be needed anymore as ``+`` / ``-`` will have a higher precedence than
+  ``~``):
+
+  Before::
```

Comment:
This is a bit confusing:

```rst
  Before::

    {{ '42' ~ 1 + 41 }}

    {# which is equivalent to #}

    {{ ('42' ~ 1) + 41 }}

  After::

    {{ '42' ~ (1 + 41) }}
```

Maybe use something like this instead?:

```rst
  {# the following expression: #}
  {{ '42' ~ 1 + 41 }}

  {# in Twig 3.x is equivalent to #}
  {{ ('42' ~ 1) + 41 }}

  {# in Twig 4.x is equivalent to #}
  {{ '42' ~ (1 + 41) }}
```

Reply:
See how I changed the example to be clearer.